Unit testing. JUnit

EugenMakarevich opened the pull request:
Task 10:

1. Develop unit tests for JsonParser class (one of them should be Exception test with >= 5 datasets). Determine by yourself, how many tests should be developed

2. Develop 1 test for RealItem class

3. Develop 1 test for VirtualItem class

4. Develop 2 tests for Cart class

5. Disable one of tests from point 1

*6. Add grouped assertion for one of tests

Tests should be grouped in different categories. Usage of Before/After annotations is required. Use the latest version of JUnit 5. After code review we will have skype-call, where you will explain, why you decided to write directly these tests, but not another.

EugenMakarevich committed:
Unit testing. JUnit

Task 10:

1. Develop unit tests for JsonParser class (one of them should be Exception test with >= 5 datasets). Determine by yourself, how many tests should be developed

2. Develop 1 test for RealItem class

3. Develop 1 test for VirtualItem class

4. Develop 2 tests for Cart class

5. Disable one of tests from point 1

*6. Add grouped assertion for one of tests

Tests should be grouped in different categories. Usage of Before/After annotations is required. Use the latest version of JUnit 5. After code review we will have skype-call, where you will explain, why you decided to write directly these tests, but not another.

andrei-unuk commented on src/test/java/unit/JsonParserTest.java:
```diff
@@ -0,0 +1,66 @@
+        parser.writeToFile(expectedCart);
+
+        //Newly created JSON file
+        File json = new File("src/main/resources/" + expectedCart.getCartName() + ".json");
```
I suggest to cearte a constant for "src/main/resources/" because you are using this path in multiple tests.

andrei-unuk replied:
It is not critical but good practice is to use String.format instead of concatenation.

andrei-unuk commented on src/test/java/unit/CartTest.java:
```diff
@@ -0,0 +1,43 @@
+    private static Cart cart;
+    private static final double TAX = 0.2;
+
+    @BeforeAll
```
According requirements you should also use 'After' annotation. Please add it (for example to remove test files or so)

EugenMakarevich committed:
Working on code review comments. Changes:
1. Added constant for "src/main/resources/".
2. Applied String.format for File json.
3. Added @Before and @AfterAll annotation for all tests
